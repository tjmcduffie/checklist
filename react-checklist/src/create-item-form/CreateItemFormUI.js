import classnames from 'classnames';
import CreateItemButtonUI from './CreateItemButtonUI';
import CreateItemInputUI from './CreateItemInputUI';
import React from 'react';
import '../util/Flex.css';
import '../util/Spacing.css';

const CreateItemFormUI = ({onChange, onSubmit, value}) => (
  <form
    className={classnames('flex-row')}
    onSubmit={onSubmit}>
    <CreateItemInputUI
      className={classnames('flex-grow', 'space-after-regular')}
      onChange={onChange}
      value={value}
    />
    <CreateItemButtonUI className="flex-shrink" />
  </form>
);

export default CreateItemFormUI;
