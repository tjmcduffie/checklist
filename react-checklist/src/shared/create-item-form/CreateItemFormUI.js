import classnames from 'classnames';
import CreateItemButtonUI from './CreateItemButtonUI';
import CreateItemInputUI from './CreateItemInputUI';
import React from 'react';

import '../base/css/Flex.css';
import '../base/css/Spacing.css';

const CreateItemFormUI = ({onChange, onSubmit, value}) => (
  <form
    className={classnames('flex-row')}
    onSubmit={onSubmit}>
    <CreateItemInputUI
      className={classnames('flex-grow', 'space-after-regular')}
      onChange={onChange}
      value={value}
    />
    <CreateItemButtonUI className="flex-shrink" />
  </form>
);

export default CreateItemFormUI;
