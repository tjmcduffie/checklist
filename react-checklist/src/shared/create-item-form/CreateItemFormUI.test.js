import CreateItemFormUI from './CreateItemFormUI';
import React from 'react';
import ReactDOM from 'react-dom';
import {create} from 'react-test-renderer';

describe('CreateItemFormUI', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <CreateItemFormUI onChange={noop} onSubmit={noop} value="" />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders predictably', () => {
    const elem = create(<CreateItemFormUI onChange={noop} onSubmit={noop} value="" />);
    expect(elem.toJSON()).toMatchSnapshot();
  });
});
