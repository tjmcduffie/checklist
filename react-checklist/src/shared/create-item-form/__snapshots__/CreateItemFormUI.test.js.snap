// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateItemFormUI renders predictably 1`] = `
<form
  className="flex-row"
  onSubmit={[Function]}
>
  <input
    className="create-item-input flex-grow space-after-regular"
    id="description"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="grey-button flex-shrink"
    type="submit"
  >
    Add
  </button>
</form>
`;
